pkg/apis/serving/k8s_validation.go

```diff
@@ -267,7 +270,85 @@ func ValidatePodSpec(ps corev1.PodSpec) *apis.FieldError {
 	return errs
 }
 
+func validateContainers(ctx context.Context, containers []corev1.Container, volumes sets.String) *apis.FieldError {
+	var errs *apis.FieldError
+	cfg := config.FromContextOrDefaults(ctx).Defaults
+	if !cfg.EnableMultiContainer {
+		errs = errs.Also(&apis.FieldError{Message: fmt.Sprintf("enable-multi-container is off, "+
+			"but found %d containers", len(containers))})
+	} else {
+		errs = errs.Also(validateContainersPorts(containers).ViaField("containers"))
+		for i := range containers {
+			// Probes are not allowed on other than serving container,
+			// ref: http://bit.ly/probes-condition
+			if len(containers[i].Ports) == 0 {
+				errs = errs.Also(validateSidecarContainer(containers[i], volumes).ViaFieldIndex("containers", i))
+			} else {
+				errs = errs.Also(ValidateContainer(containers[i], volumes).ViaFieldIndex("containers", i))
+			}
+		}
+	}
+	return errs
+}
+
+// validateContainersPorts validates port when specified multiple containers
+func validateContainersPorts(containers []corev1.Container) *apis.FieldError {
+	var (
+		count int
+		errs  *apis.FieldError
+	)
+	for i := range containers {
+		count += len(containers[i].Ports)
+	}
+	// When no container ports are specified.
+	if count == 0 {
+		errs = errs.Also(apis.ErrMissingField("ports"))
+	}
+	// More than one container sections have ports.
+	if count > 1 {
+		errs = errs.Also(apis.ErrMultipleOneOf("ports"))
+	}
+	return errs
```

Comment:
```suggestion
	return nil
```

and ditch the variable if you adopt the other suggestions.